Initizalize test framework

julkunensuvi opened the pull request:
Decided to include only unit testing. Chose Jest as a tool

rafinutshaw commented on package.json:
```diff
@@ -8,20 +8,24 @@
-    "@types/uuid": "^9.0.8",
-    "typescript": "^5.2.2",
+    "ts-jest": "^29.1.2",
+    "typescript": "^5.4.2",
```
Why update the typescript version?
Because of this update. I am getting this warning when i run "npm run lint"
![image](https://github.com/lightsheet-team/lightsheet/assets/24876640/77bf2148-0838-4d3d-8eb4-638109266a7d)

julkunensuvi replied:
Left the old version!

rafinutshaw commented on tests/core/structure/sheet.test.ts:
```diff
@@ -0,0 +1,24 @@
+
+    sheet.setCellAt(colPos, rowPos, value);
+
+    //const positionInfo = sheet.setCellAt(colPos, rowPos, value);
```
Kindly remove any commented codes

julkunensuvi replied:
Done

rafinutshaw reviewed (changes requested):
Kindly add these tests to github actions as well. Github actions shouldnt allow merge if tests fail.

rafinutshaw reviewed (changes requested):
Getting this warning when i run "npm run test". Kindly looking to this as well

![image](https://github.com/lightsheet-team/lightsheet/assets/24876640/8c6145e6-aa19-437c-8702-581a43218d08)

rafinutshaw commented:
This branch has is not updated with the latest main branch. Kindly update with the main branch and resolve conflicts.
Thanks.

julkunensuvi commented:
Fixed the issues and added testing to the github workflow. Linked the github actions issue to the PR as well 
https://github.com/lightsheet-team/lightsheet/issues/11
 @rafinutshaw @GimanthaDissanayake

rafinutshaw commented:
LGTM!
